### ISSUE TITLE
[BUG]: Volt is defined and array in 

### ISSUE THREAD

**Issue by ismail0234** [phalcon/cphalcon#14816]
Questions? Forum: https://phalcon.link/forum or Discord: https://phalcon.link/discord

**Describe the bug**
Using these two together will give the wrong result.

> Provide minimal script to reproduce the issue

```volt
// WRONG Result:
{% if categoryIds is defined AND category['id'] in categoryIds %}
    checked
{% endif %}

// Correct Result:
{% if categoryIds is defined AND (category['id'] in categoryIds) %}
    checked
{% endif %}
```

**Expected behavior**

```php
// WRONG RESULT
<?php if ($this->isIncluded(isset($categoryIds) && $category['id'], $categoryIds)) { ?>
   checked
<php } ?>

// Correct RESULT
<?php if (isset($categoryIds) && $this->isIncluded($category['id'], $categoryIds)) { ?>
   checked
<php } ?>
```

**Screenshots**

**Details**
 - Phalcon version: v3.4.5
 - PHP Version: 7.3
 - Operating System: Windows
 - Installation type: installing via package manager
 - Zephir version (if any): Latest
 - Server: FASTCGI

**Comment by Jeckerson** [comment]
Confirmed, here is screenshot of generated code from volt

![image](https://user-images.githubusercontent.com/3289702/74016613-9dcd1880-498a-11ea-8f1b-b92d9b7f7d68.png)

**Comment by niden** [comment]
This needs work on the PHQL parser since the right part after the `&&` is not parsed on its own.

For now the workaround with parentheses works but the suggested syntax should work also.

**Comment by sergeyklay** [comment]
Fixed in 4.0.x branch. Thank you for the report, and for helping us make Phalcon better.
